endpoints/apiconfig.go

```diff
@@ -757,16 +757,35 @@ func validateRequest(r interface{}) error {
 
 	t := v.Type()
 	for i := 0; i < v.NumField(); i++ {
-		if err := validateField(v.Field(i), t.Field(i)); err != nil {
+		if err := validateField(v.Field(i), t.Field(i), t); err != nil {
 			return err
 		}
 	}
 	return nil
 }
 
-func validateField(v reflect.Value, t reflect.StructField) error {
-	// only validate simple types, ignore arrays, slices, chans, etc.
-	if v.Kind() > reflect.Float64 && v.Kind() != reflect.String {
+func hasExportedFields(v reflect.Value) bool {
+	t := v.Type()
+	n := t.NumField()
+
+	if n == 0 {
+		return false
+	}
+
+	for i := 0; i < n; i++ {
+		if t.Field(i).PkgPath == "" {
+			return true
+		}
+	}
+
+	return false
+}
+
+func validateField(v reflect.Value, t reflect.StructField, parent reflect.Type) error {
```

Comment:
I passed in the parent because I added `return fmt.Errorf("missing field %v in %v", t.Name, parent.Name())` a parent to a missing field error. 
I did it because if you have multiple fields with the same name in a struct and nested structs, the default `missing field %v` felt non descriptive, you wouldn't know which struct it belongs to. But if you feel like it's too much, I will remove it :)